drone-localization/core/services/evaluation/metrics.py

```
@@ -0,0 +1,83 @@
+    indices = torch.argsort(dist_matrix, dim=1, descending=True)
+
+    recall_1, recall_5, recall_10 = 0, 0, 0
+    all_ap = []
+
+    for i in range(len(query_labels)):
+        label = query_labels[i]
+        ranked_labels = gallery_labels[indices[i]]
+        correct_pos = (ranked_labels == label).nonzero(as_tuple=False).flatten()
+        if len(correct_pos) > 0:
+            first_match = correct_pos[0].item()
+            if first_match < 1: recall_1 += 1
```

Ruff (0.15.12)

[error] Multiple statements on one line (colon)

(E701)